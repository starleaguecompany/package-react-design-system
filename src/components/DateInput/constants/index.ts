export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
