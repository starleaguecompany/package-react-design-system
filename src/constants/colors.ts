export const COLORS = {
  GRAY: 'gray',
  GREEN: 'green',
  BLUE: 'blue',
  ORANGE: 'orange',
  VIOLET: 'violet',
  RED: 'red',
} as const
