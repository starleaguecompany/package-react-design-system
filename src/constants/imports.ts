export const reactDSImportPath = '@starleaguecompany/react-design-system';
