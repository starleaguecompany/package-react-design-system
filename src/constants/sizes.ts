export const CONTENT_SIZES = {
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
} as const;

export const CONTAINER_SIZES = {
  S36: 36,
  S44: 44,
  S52: 52,
  S60: 60,
} as const;

export const FONT_SIZES = {
  S10: 10,
  S12: 12,
  S14: 14,
  S16: 16,
  S20: 20,
} as const;
