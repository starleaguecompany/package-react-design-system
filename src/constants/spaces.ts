export const SPACE_SIZES = {
  S0: 0,
  S2: 2,
  S4: 4,
  S8: 8,
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
  S32: 32,
  S36: 36,
  S40: 40,
} as const;
