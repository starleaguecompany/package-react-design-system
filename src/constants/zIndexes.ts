export const Z_INDEXES = {
  DROPDOWN: 1060,
  STICKY: 1020,
  FIXED: 1030,
  MODAL_BACKDROP: 1040,
  MODAL: 1050,
  TOOLTIP: 1060,
} as const;
