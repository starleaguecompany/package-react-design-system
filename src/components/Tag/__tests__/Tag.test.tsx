import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { CONTAINER_SIZES } from '../../../constants/sizes';
import { Menu } from '@starleaguecompany/react-icons';

import { Tag, Icon } from '../../../index';
import userEvent from '@testing-library/user-event';

describe('Tag', () => {
  test('should render correctly with no props', () => {
    render(<Tag />);

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    expect(screen.getByTestId('Tag')).toHaveClass('container size-44', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<Tag id="test-id" className="test-class" />);

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    expect(screen.getByTestId('Tag')).toHaveAttribute('id', 'test-id');
    expect(screen.getByTestId('Tag')).toHaveClass('test-class');
  });

  test('should render correctly with text', () => {
    render(<Tag>Testing tag</Tag>);

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    expect(screen.getByText('Testing tag')).toBeVisible();
  });

  test('should render correctly with size prop', () => {
    render(<Tag size={CONTAINER_SIZES.S36}>Testing tag</Tag>);

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    expect(screen.getByTestId('Tag')).toHaveClass('size-36');
  });

  test('should render correctly with active prop', () => {
    render(<Tag active>Testing tag</Tag>);

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    expect(screen.getByTestId('Tag')).toHaveClass('active');
  });

  test('should render correctly with closable prop and execute onClose callback on click', () => {
    const closeCallback = jest.fn();
    render(
      <Tag closable onClose={closeCallback}>
        Testing tag
      </Tag>
    );

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    expect(screen.getByTestId('Tag')).toContainElement(screen.getByTestId('Cross'));
    userEvent.click(screen.getByTestId('Cross'));
    expect(closeCallback).toBeCalledTimes(1);
  });

  test('should render correctly with icon', () => {
    render(
      <Tag>
        <Icon icon={<Menu />} />
        Testing pane
      </Tag>
    );

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    expect(screen.getByTestId('Menu')).toBeVisible();
    expect(screen.getByText('Testing pane')).toBeVisible();
    expect(screen.getByTestId('Tag')).toContainElement(screen.getByTestId('Menu'));
  });

  test('should render correctly on external value update', () => {
    const onChangeCallback = jest.fn();
    const { rerender } = render(<Tag active={false} onChange={onChangeCallback} />);

    expect(screen.getByTestId('Tag')).toMatchSnapshot();
    rerender(<Tag active={true} onChange={onChangeCallback} />);
    expect(screen.getByTestId('Tag')).toHaveClass('active');
    expect(onChangeCallback).toBeCalledTimes(0);
  });
});
