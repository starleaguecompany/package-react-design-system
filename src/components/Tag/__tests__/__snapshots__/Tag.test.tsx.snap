// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render correctly on external value update 1`] = `
<span
  class="container size-44"
  data-qa="Tag"
/>
`;

exports[`Tag should render correctly with active prop 1`] = `
<span
  class="container active size-44"
  data-qa="Tag"
>
  Testing tag
</span>
`;

exports[`Tag should render correctly with attributes 1`] = `
<span
  class="test-class container size-44"
  data-qa="Tag"
  id="test-id"
/>
`;

exports[`Tag should render correctly with closable prop and execute onClose callback on click 1`] = `
<span
  class="container size-44"
  data-qa="Tag"
>
  Testing tag
  <span
    class="container size-20 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Cross"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0)"
      >
        <path
          d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
          stroke="currentColor"
          stroke-dasharray="10 5"
        />
      </g>
      <defs>
        <clippath
          id="clip0"
        >
          <path
            d="M0 0h20v20H0z"
            fill="currentColor"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</span>
`;

exports[`Tag should render correctly with icon 1`] = `
<span
  class="container size-44"
  data-qa="Tag"
>
  <span
    class="container size-20 shape-round"
    data-qa="Icon"
  >
    <svg
      data-qa="Menu"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 5h12M4 10h12M4 15h12"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </span>
  Testing pane
</span>
`;

exports[`Tag should render correctly with no props 1`] = `
<span
  class="container size-44"
  data-qa="Tag"
/>
`;

exports[`Tag should render correctly with size prop 1`] = `
<span
  class="container size-36"
  data-qa="Tag"
>
  Testing tag
</span>
`;

exports[`Tag should render correctly with text 1`] = `
<span
  class="container size-44"
  data-qa="Tag"
>
  Testing tag
</span>
`;
